### PR TITLE
[OUTDATED] namcoic.cpp, namcos2.cpp, namconb1.cpp, namcofl.cpp Cleanups

### PR DESCRIPTION
**Pull request by cam900**
namcoic.cpp : Cleanup Sprite sizing for metlhawk*, Move sprite swap xy function into sprite draw routine, Cleanup duplicates, Add notes

namcos2.cpp : Move sgunner2 specific m_player_mux into namcos2_state, Move c116 emulation into device, Cleanup gfxdecode, bankswitches, unnecessary things, others
(Reduce sprite gfxdecode, because spriterom bank size is differs per game(ex.Four Trax has 4 of 256Kx32bit spriterom bank, etc...)), Improve region names

namconb1.cpp, namcofl.cpp : Fix gfxdecode name, Cleanup gfxdecode, Minor cleanups

namcofl.cpp : Reduce runtime tag lookups, Minor cleanups

namco_c116.cpp : Minor cleanup

namcos2.cpp, namcoic.cpp, namconb1.cpp : Cleanup bitswaps/bitshifts

### CODE REVIEW

File: src/mame/drivers/namcos2.cpp [hunk 1]
```diff
@@ -5727,8 +5579,8 @@ DRIVER_INIT_MEMBER(namcos2_state,marvland)
 DRIVER_INIT_MEMBER(namcos2_state,metlhawk)
 {
 	/* unscramble sprites */
-	uint8_t *data = memregion("gfx1")->base();
-	for (int i=0; i<0x200000; i+=32*32)
+	uint8_t *data = memregion("sprites")->base();
+	for (int i=0; i<memregion("sprites")->bytes(); i+=32*32)
```
Comment (cuavas) [inline src/mame/drivers/namcos2.cpp]: Please at least lift the call to `memregion("sprites")->bytes()` out of the loop - right now you're doing a tagmap lookup when it checks the condition on every iteration of the loop.

**Comment by cam900** [comment]
It's for make some simpler gfxdecode, sound bankswitching, and tilemap code descrambling.
and sprite number functions of metal hawk video board sprite system is similar standard namco system 2 boards.

**Comment by cuavas** [comment]
The thing is, we're not supposed to be messing with ROM loading just for gfxdecode convenience - the ROM loading should represent how the data is addressed by the system.  If you can show that the system actually sees the ROMs in this order with address line swaps or something, I'm fine with the change.  Otherwise, the gfxdecode should match how the system maps ROMs, not the other way around.

**Comment by galibert** [comment]
Note that a good clue, when it exists, is how the game addresses the roms
for checksumming.  It's usually very significant about how they're wired
up.  Of course, video rom checksumming doesn't happen that often outside of
Konami...

  OG.

**Comment by cam900** [comment]
In assault (namco system 2) schematics, ROM Data bus is connected on OCH0-7(5B/5C), OCH8-15(4B/4C), OCH16-23(5D/5F), OCH24-31(4D/4F).

**Comment by thepunisher007** [comment]
Hyperneogeo64 when?

File: src/mame/machine/namcoic.cpp [hunk 2]
```diff
@@ -121,46 +112,49 @@ void namcos2_shared_state::c123_SetTilemapControl(int offset, uint16_t newword)
...
-	case 0x1a/2:
-		m_c123_TilemapInfo.tmap[3]->set_scrollx( 0, newword );
-		break;
-	case 0x1e/2:
-		m_c123_TilemapInfo.tmap[3]->set_scrolly( 0, newword );
-		break;
+		if ((offset & 7) < 6)
+		{
+			if (offset & 8) // 30-3f Color
+				m_c123_TilemapInfo.tmap[offset & 7]->set_palette_offset( 0, (newword&7)*256 );
```
Comment (Osso13) [inline src/mame/machine/namcoic.cpp]: this doesn't compile: set_palette_offset only takes one argument

File: src/mame/machine/namcoic.cpp [hunk 3]
```diff
@@ -488,38 +478,23 @@ void namcos2_state::draw_sprites_metalhawk(screen_device &screen, bitmap_ind16 &
...
-				rect.max_x += (tile&1)?16:0;
-				rect.min_y += (tile&2)?16:0;
-				rect.max_y += (tile&2)?16:0;
+				scalex >>= 1; // TODO : Correct at everywhere?
+				gfx->set_source_clip((tile&0x0001) ? 16 : 0, 16, (tile&0x0002) ? 16 : 0, 16);
 			}
 			zdrawgfxzoom(
 				screen,
 				bitmap,
 				rect,
```
Comment (Osso13) [inline src/mame/machine/namcoic.cpp]: you're removed rect  everywhere but here, so it won't compile

**Comment by cuavas** [comment]
If this can be updated for current source, and tested to ensure it doesn't break stuff, I guess it can be approved.  Cleaning up the ROM loading for audio CPUs is good at any rate, and the ROM loading isn't particularly clean as-is.

**Comment by cuavas** [comment]
How well have you tested this?  There have been some rather obvious issues with some of your recent pull requests, for example the title screen not working in Final Lap.  This pull request affects many systems, so it needs to be tested properly.

**Comment by cam900** [comment]
Tested, finallap is now bootable. but title screen is still wrong.
GFX is almost same.

**Comment by cam900** [comment]
This PR is now outdated, I close for this.

**Pull request closed without merging**